### PR TITLE
fix(urls): use path and path converters

### PR DESCRIPTION
**Pull request by 0xMRTT**
https://docs.djangoproject.com/en/4.1/topics/http/urls/#path-converters

### CODE REVIEW

File: avatar/urls.py [hunk 1]
```diff
@@ -1,18 +1,23 @@
...
+    path("delete/", views.delete, name="avatar_delete"),
+    # https://docs.djangoproject.com/en/4.1/topics/http/urls/#path-converters
+    path(
+        "render_primary/<slug:user>/<int:width>/",
         views.render_primary,
         name="avatar_render_primary",
     ),
     re_path(
-        r"^render_primary/(?P<user>[\w\d\@\.\-_]+)/(?P<width>[\d]+)/(?P<height>[\d]+)/$",
+        "render_primary/<slug:user>/<int:width>/<int:height>",
```
Comment (johanneswilm) [inline avatar/urls.py:20]: @0xMRTT Is there a particular reason why we keep `re_path` here? And it looks like we remove the `/` at the end?

Comment (0xMRTT) [reply avatar/urls.py:20]: Oh my bad, missed to remove it 🙃

File: avatar/templatetags/avatar_tags.py [hunk 2]
```diff
@@ -94,7 +94,7 @@ def primary_avatar(user, width=settings.AVATAR_DEFAULT_SIZE, height=None):
     else:
         kwargs["height"] = height
 
-    url = reverse("avatar_render_primary", kwargs=kwargs)
+    url = reverse("avatar:render_primary", kwargs=kwargs)
```
Comment (johanneswilm) [inline avatar/templatetags/avatar_tags.py:97]: changing the reverse urls probably means that we will need to release it as part of a new major version number.

Comment (0xMRTT) [reply avatar/templatetags/avatar_tags.py:97]: Yes, but now i need to solve tests :D

Comment (johanneswilm) [reply avatar/templatetags/avatar_tags.py:97]: Sure. If we make a new major release, we probably need to add other things as well. Or we just keep the PR open for now if there is nothing else we want to change/fix right now.

**Pull request merged**